classes/profile.php

```diff
@@ -194,8 +194,9 @@ public function save_record(): int {
         $this->raw_set('usermodified', $USER->id);
 
         // We may not have obtained a valid userid when the profile record was created.
-        // If the current profile userid is 0, but there's a valid $USER->id, update it.
-        if ($USER->id && ! $this->raw_get('userid')) {
+        // If the current $USER->id is non-zero, update it unless the profile userid is admin.
+        $profileuser = (int) $this->raw_get('userid');
+        if ($USER->id && $profileuser !== 1) {
```

Comment:
What is the back story here? 1 looks like a magic number to me, perhaps instead you should be leveraging is_guest() ?

Reply:
I should have asked but I thought I remembered that admin is always userid 1 in Moodle. I'll look for a call to find the admin uid.

Reply:
Google says 2, so I'll look into this deeper and correct it.

Reply:
Looks like guest is 1 and admin is 2. Now, I just need to find out if that's always the case, or how to find the admin userid if it's not.

Reply:
The admin id can be anything and can change over time. But I would like to understand the intent of the code first and that should be reflected in the comment, and then worry about the details